data/campaigns/tutorial03_seafaring.wmf/scripting/mission_thread.lua

```diff
@@ -83,24 +100,58 @@ function expedition()
       end
    end
 
-   conclude()
+   conclude_expedition()
 end
 
-function conclude()
+function conclude_expedition()
    additional_port_space.terr = "desert_steppe" -- make it land again so that the player can build a port
-   message_box_objective(plr, conclusion)
-   sleep(5000)
-   waterways()
+   message_box_objective(plr, expedition4)
+   sleep(3000)
+   iron_mine()
+end
+
+function iron_mine()
+   local o = message_box_objective(plr, expedition5)
+   while #plr:get_buildings("atlanteans_ironmine") < 1 do sleep(3000) end
+   set_objective_done(o)
+   expedition_done = true
 end
 
 function complete_ferry_yard()
    plr:allow_buildings{"atlanteans_ferry_yard"}
    local o = message_box_objective(plr, ferry_2)
-   while #plr:get_buildings("atlanteans_ferry_yard") < 1 do sleep(200) end
+   while #plr:get_buildings("atlanteans_ferry_yard") < 1 do sleep(3000) end
    set_objective_done(o)
 end
 
+function complete_gold_mine()
+   local field_for_mine = map:get_field(20, 102):region(5)
```

Comment:
Rename this variable to `fields_for_mine` so that it's clear from the name that it's a region and not a single field.